### PR TITLE
Windows Compatibility

### PR DESCRIPTION
**Pull request by nicolsen**
Dan's Windows compatibility changes.
* [x] Verify that demos run to completion
* [x] Pull develop into feature/granular-dynamics
* [x] Format files
* [x] Review code changes
* [ ] Squash on merge

### CODE REVIEW

File: src/chrono_granular/physics/ChGranular.cpp [hunk 1]
```diff
@@ -172,10 +172,11 @@ void ChSystemGranularSMC::packSphereDataPointers() {
...
 
-    // force prefetch the sphere data pointer after update
-    int dev_ID;
-    gpuErrchk(cudaGetDevice(&dev_ID));
-    gpuErrchk(cudaMemPrefetchAsync(sphere_data, sizeof(*sphere_data), dev_ID));
+    // DN: had to comment this prefetch for now; crashing on Windows
+    //// force prefetch the sphere data pointer after update:
+    // int dev_ID;
+    // gpuErrchk(cudaGetDevice(&dev_ID));
+    // gpuErrchk(cudaMemPrefetchAsync(sphere_data, sizeof(*sphere_data), dev_ID));
```
Comment (nicolsen) [inline src/chrono_granular/physics/ChGranular.cpp]: This is a concern if we care about multi-GPU

File: src/demos/granular/demo_GRAN_ShearBand.cpp [hunk 2]
```diff
@@ -54,6 +55,12 @@ int main(int argc, char* argv[]) {
     // Setup simulation
     ChSystemGranularSMC gran_sys(params.sphere_radius, params.sphere_density,
                                  make_float3(params.box_X, params.box_Y, params.box_Z));
+
+	// to do: don't expose the guts of granular at this level; work through an API
+	// but for now get it going like this
```
Comment (nicolsen) [inline src/demos/granular/demo_GRAN_ShearBand.cpp]: Dan, looks like you still have a TODO?

File: src/demos/granular/demo_GRAN_fixedterrain.cpp [hunk 3]
```diff
@@ -47,13 +48,16 @@ int main(int argc, char* argv[]) {
         return 1;
     }
 
-    // Chrono::Granular system
     ChSystemGranularSMC gran_sys(params.sphere_radius, params.sphere_density,
                                  make_float3(params.box_X, params.box_Y, params.box_Z));
 
+	// to do: don't expose the guts of granular at this level; work through an API
+    // but for now get it going like this
```
Comment (nicolsen) [inline src/demos/granular/demo_GRAN_fixedterrain.cpp]: @DanNegrut TODO

File: src/demos/granular/demo_GRAN_terrainBox_SMC.cpp [hunk 4]
```diff
@@ -68,6 +69,12 @@ int main(int argc, char* argv[]) {
     // Setup simulation
     ChSystemGranularSMC gran_sys(params.sphere_radius, params.sphere_density,
                                  make_float3(params.box_X, params.box_Y, params.box_Z));
+
+    // to do: don't expose the guts of granular at this level; work through an API
+    // but for now get it going like this
```
Comment (nicolsen) [inline src/demos/granular/demo_GRAN_terrainBox_SMC.cpp]: @DanNegrut TODO

File: src/chrono_granular/physics/ChGranular.cpp (hunk 1 above)
Comment (DanNegrut) [reply src/chrono_granular/physics/ChGranular.cpp]: Nic - let's focus on validation for now and not worry for the time being about the multi-GPU.

**Pull request merged**